impeller/compiler/spirv_sksl.cc

```diff
@@ -219,47 +220,14 @@ bool CompilerSkSL::emit_uniform_resources() {
   bool emitted = false;
 
   // Output Uniform Constants (values, samplers, images, etc).
-  std::vector<ID> regular_uniforms;
-  std::vector<ID> shader_uniforms;
-  for (auto& id : ir.ids) {
-    if (id.get_type() == TypeVariable) {
-      auto& var = id.get<SPIRVariable>();
-      auto& type = get<SPIRType>(var.basetype);
-      if (var.storage != StorageClassFunction && !is_hidden_variable(var) &&
-          type.pointer &&
-          (type.storage == StorageClassUniformConstant ||
-           type.storage == StorageClassAtomicCounter)) {
-        // Separate out the uniforms that will be of SkSL 'shader' type since
-        // we need to make sure they are emitted only after the other uniforms.
-        if (type.basetype == SPIRType::SampledImage) {
-          shader_uniforms.push_back(var.self);
-        } else {
-          regular_uniforms.push_back(var.self);
-        }
-        emitted = true;
-      }
-    }
+  std::vector<ID> regular_uniforms =
```

Comment:
The revert was required because this change was filtering out non float uniforms, which we would never emit, and therefore never error check. We could move the error checking earlier in the pipeline, now that I'm more familiar with the API that is probably a better place to do it.

That said for now, the minimal change is to sort all unfiorms except sampled images instead of all float uniforms.